File: trunk/sources/modules/client/desktop/desktop-utils/src/main/java/eu/scy/client/desktop/desktoputils/log4j/InitLog4j.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package eu.scy.client.desktop.desktoputils.log4j;

import java.net.URL;
import org.apache.log4j.xml.DOMConfigurator;

/**
 *
 * @author sikkenj
 */
public class InitLog4j
{

   public static void init(URL configUrl)
   {
      try
      {
         // System.out.println("reading log4j config from " + configUrl);
         DOMConfigurator.configure(configUrl);
      }
      catch (Exception e)
      {
         // System.out.println("Problems with loading log4j config, from " + configUrl);
         e.printStackTrace();
      }

   }

   public static void initFromClassPath(String configFileName)
   {
      URL configUrl = new Object().getClass().getResource(configFileName);
      init(configUrl);
   }

   public static void init()
   {
      initFromClassPath("/config/scy-lab-log4j.xml");
   }
}

File: trunk/sources/external/missionCopier/src/main/scala/eu/scy/utils/missioncopier/InitLog4J.scala
package eu.scy.utils.missioncopier

import org.apache.log4j.xml.DOMConfigurator

/**
 * Created by IntelliJ IDEA.
 * User: SikkenJ
 * Date: 29-11-11
 * Time: 16:30
 * To change this template use File | Settings | File Templates.
 */

object InitLog4J {
  def initLog4J(fileName: String) {
    try {
      val configUrl = this.getClass().getResource(fileName)
      if (configUrl != null) {
        println("reading log4j config from " + configUrl);
        DOMConfigurator.configure(configUrl);
      } else {
        println("cannot find log4j configuration file: " + fileName);
      }

    } catch {
      case e: Exception => println("Problems with loading log4j config, from " + fileName)
      e.printStackTrace()
    }
  }

}